src/celeritas/CMakeLists.txt

```diff
@@ -153,33 +153,28 @@ if(CELERITAS_USE_ROOT)
   # functions set this to the same value for our targets.
   set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CELERITAS_LIBRARY_OUTPUT_DIRECTORY})
 
-  # Generate the dictionary source file
-  root_generate_dictionary(CeleritasRootInterface
-    "${CMAKE_CURRENT_SOURCE_DIR}/io/ImportData.hh"
-    "${CMAKE_CURRENT_SOURCE_DIR}/io/ImportElement.hh"
-    "${CMAKE_CURRENT_SOURCE_DIR}/io/ImportMaterial.hh"
-    "${CMAKE_CURRENT_SOURCE_DIR}/io/ImportParticle.hh"
-    "${CMAKE_CURRENT_SOURCE_DIR}/io/ImportPhysicsTable.hh"
-    "${CMAKE_CURRENT_SOURCE_DIR}/io/ImportPhysicsVector.hh"
-    "${CMAKE_CURRENT_SOURCE_DIR}/io/ImportProcess.hh"
-    "${CMAKE_CURRENT_SOURCE_DIR}/io/ImportVolume.hh"
-    NOINSTALL
-    MODULE celeritas
-    LINKDEF "${CMAKE_CURRENT_SOURCE_DIR}/ext/RootInterfaceLinkDef.h"
-  )
   celeritas_add_object_library(celeritas_root
     ext/RootExporter.cc
     ext/RootImporter.cc
     ext/ScopedRootErrorHandler.cc
     ext/detail/RootUniquePtr.root.cc
     io/RootFileManager.cc
     io/RootStepWriter.cc
-    "${CMAKE_CURRENT_BINARY_DIR}/CeleritasRootInterface.cxx"
   )
 
   target_link_libraries(celeritas_root
     PRIVATE Celeritas::corecel ROOT::Core ROOT::Tree
   )
+  # reflex_generate_dictionary will only look at the include directories if
+  # its first argument is a target (and it does not recurse to the target's
+  # dependencies)
+  target_include_directories(celeritas_root PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
+  reflex_generate_dictionary(celeritas_root
+    "${CMAKE_CURRENT_SOURCE_DIR}/ext/RootInterfaceClasses.hh"
+    SELECTION "${CMAKE_CURRENT_SOURCE_DIR}/ext/RootInterface_sel.xml"
+    OPTIONS "-llibceleritas.so"  # equivalent to root_generate_dictionary MODULE option
```

Comment:
Will this work on mac where it makes a dylib? What about with static libs?

Reply:
Apriori yes.  It is used only as a stem to derive names.  (for better or worse, the `.so` is actually 'required' by the current code!

Reply:
Maybe it would be good to define a celeritas helper function that hides the bizarre option requirements if we need to reuse it in the demo app for I/O as well...

Reply:
Indeed.  `reflex_generate_dictionary` is significantly under-features compare to `root_generate_dictionary`.

Reply:
BTW I wrote that comment before our meeting when you suggested the same thing 😄